localstripe/resources.py

```diff
@@ -736,6 +736,39 @@ def _api_list_tax_ids(cls, id, **kwargs):
         obj = cls._api_retrieve(id)
         return obj.tax_ids
 
+    @classmethod
+    def _api_add_subscription(cls, id, **data):
+        obj = Subscription(customer=id, **data)
+        return obj
+
+    @classmethod
+    def _api_retrieve_subscription(cls, id, subscription_id, **kwargs):
+        if kwargs:
+            raise UserError(400, 'Unexpected ' + ', '.join(kwargs.keys()))
+
+        obj = Subscription._api_retrieve(subscription_id)
+
+        if obj.customer != id:
+            raise UserError(400, 'Customer ' + id + ' does not have a subscription with ID ' + subscription_id)
```

Comment:
Isn't it a 404 instead?

If this is the case, a `UserError(404, 'Not Found')` should be enough (and will suppress the linting error).

Reply:
Yeah you're right, it's a 404. I can change that. However I thought it'd be nice to have the same error message as the live api which is why I included the actual message.
Unless you'd still rather just have the generic `UserError(404, 'Not Found')`, though could just as well relate to the customer itself (?)  
![image](https://user-images.githubusercontent.com/8465957/68849838-177aa080-06d3-11ea-94d3-3ee464588a34.png)

Reply:
Having the same as Stripe (404 + your original text) seems perfect :+1: